async-nats/src/connector.rs

```diff
@@ -154,6 +154,7 @@ impl Connector {
                             Authorization::None => {
                                 connection.write_op(ClientOp::Connect(connect_info)).await?;
 
+                                self.events_tx.send(Event::Reconnect).await.ok();
```

Comment:
`Reconnect` would be sent even on initial connection.

Best would be to rename this variant to `Connected`, as it does not imply previous state? @caspervonb

Reply:
So Pending, Connected, Disconnected? sgtm